src/WorkerChannel.ts

```diff
@@ -209,6 +209,11 @@ export class WorkerChannel implements IWorkerChannel {
                     data: info.outputBindings[key].converter(result.return[key])
                   });
               }
+              // returned value does not match any output bindings - pass it along
+              // this specifically is to allow the durable binding to work
+              if (!response.returnValue && response.outputData.length == 0) {
+                response.returnValue = toTypedData(result.return);
+              }

```

Comment:
This is not specific to the change you're making, but I'm trying to understand the logic of this larger code block. I see that we check on line 195 whether `return.return` is truthy. I also see on line 204 that we make the same check again. Won't that second check always return true based on what was established at line 195? In other words, could we just need to just use `else` instead of `else if (return.return)` on line 204?

More closely related to your change, I see that we set the return value only if `response.returnValue` and `response.outputData` are null/empty. What would happen if the function has a regular output binding but also returns a value? Would that cause line 215 to be skipped? I see that this code is only about return values, but is there any other situation where `response.outputData.length` might be non-zero?

Reply:
@cgillum -
> could we just need to just use else instead of else if (return.return) on line 204?

yes!! changing, thanks!

> What would happen if the function has a regular output binding but also returns a value?

There's [three documented ways](https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-node#outputs) of using output bindings for the node worker. The existing code handles for these three situations. We decided to treat all returned output as belonging to an output binding named $return, if it is present in the host I believe in v1: [dupe logic here](https://github.com/Azure/azure-functions-host/blob/a2a11fb1adc87d6068019b7c8b6a288c37274201/src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs#L42-L49). For a function app that also defines output bindings using `context.bindings`, the rest of the code (starting with `result.bindings`) should handle that.

The new code only applies to output that is returned but does not have a corresponding "out" binding defined in function.json. As far as I know, durable is the only extension that uses returned output without having a corresponding "out" binding in function.json. @fabiocav - do you know if that's true?

Reply:
Update on this -
I think [this code in the durable extension](https://github.com/Azure/azure-functions-durable-extension/blob/dev/src/WebJobs.Extensions.DurableTask/Bindings/OrchestrationTriggerAttributeBindingProvider.cs#L95-L99) registers an implicit "$return". 

I've looked in these repos:
https://github.com/Azure/azure-webjobs-sdk-extensions
https://github.com/Azure/azure-functions-durable-extension
https://github.com/Azure/azure-functions-eventgrid-extension
https://github.com/Azure/azure-functions-iothub-extension
https://github.com/Azure/azure-webjobs-sdk

It does look like [HTTP triggers](https://github.com/Azure/azure-webjobs-sdk-extensions/blob/dev/src/WebJobs.Extensions.Http/HttpTriggerAttributeBindingProvider.cs#L248) are also capable of an implicit "$return" value. However, from testing, as long as an output binding is explicitly defined, I don't see the implicit "$return" and returned value changing behavior, even with this PR.

However, one behavior change introduced with this PR is that when an HTTP output binding **is not** defined, we see the returned value is made into the http response. Before, it would return a "204". Now, it will return a "best effort" http object that is serialized in this buggy way: https://github.com/Azure/azure-functions-nodejs-worker/issues/228

I think we can either keep this behavior in this PR or hard-code behavior around functions with "orchestrationTrigger" bindings. Hard-coding is the safest option that guarantees no affect on HTTP :\
@fabiocav - could you weigh in?